src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -573,6 +574,165 @@ struct intersects<PointTag, TriangleTag, Point, Triangle>
   }
 };
 
+template <typename Box, typename Triangle>
+struct intersects<BoxTag, TriangleTag, Box, Triangle>
+{
+  KOKKOS_FUNCTION static constexpr bool apply(Box const &box,
+                                              Triangle const &triangle)
+  {
+    // Based on the Separating Axis Theorem
+
+    // Test bounding boxes, i.e., normals of the box
+    Box triangle_bounding_box;
+    ArborX::Details::expand(triangle_bounding_box, triangle);
+    if (!ArborX::Details::intersects(triangle_bounding_box, box))
+      return false;
+
+    // shift to origin
+    constexpr int DIM = GeometryTraits::dimension_v<Triangle>;
+    static_assert(DIM == 3,
+                  "Box-Triangle intersection only implemented in 3d!");
+    using Float = typename GeometryTraits::coordinate_type_t<Triangle>;
+    auto min_corner = box.minCorner();
+    auto max_corner = box.maxCorner();
+    auto triangle_a = triangle.a;
+    auto triangle_b = triangle.b;
+    auto triangle_c = triangle.c;
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_ab;
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_ac;
+    ExperimentalHyperGeometry::Point<DIM, Float> extents;
+    for (int i = 0; i < DIM; ++i)
+    {
+      Float const shift = -(max_corner[i] + min_corner[i]) / 2;
+      min_corner[i] += shift;
+      max_corner[i] += shift;
+      triangle_a[i] += shift;
+      triangle_b[i] += shift;
+      triangle_c[i] += shift;
+      edge_ab[i] = triangle_b[i] - triangle_a[i];
+      edge_ac[i] = triangle_c[i] - triangle_a[i];
+      extents[i] = (max_corner[i] - min_corner[i]) / 2;
+    }
+
+    // Test normal of the triangle
+    ExperimentalHyperGeometry::Point<DIM, Float> normal{
+        {edge_ab[1] * edge_ac[2] - edge_ab[2] * edge_ac[1],
+         edge_ab[2] * edge_ac[0] - edge_ab[0] * edge_ac[2],
+         edge_ab[0] * edge_ac[1] - edge_ab[1] * edge_ac[0]}};
```

Comment:
This is a cross product, make it a helper function.

Reply:
I would only use it once, though.

Reply:
Don't you use cross products in all the tests and then just ignore a single coordinate? You could make cross product work in both 2D and 3D, and then convert things to 2D for the last 9 tests.

Reply:
> Don't you use cross products in all the tests and then just ignore a single coordinate?

Yes.

> You could make cross product work in both 2D and 3D, and then convert things to 2D for the last 9 tests.

Not quite sure what you mean. I could of course implement a generic cross-product and dot product and use that. Then I will just not be able to take advantage of all the 0s that are present. Not sure how to convert things from 3D to 2D for these tests since the zero component is different between all of these checks.